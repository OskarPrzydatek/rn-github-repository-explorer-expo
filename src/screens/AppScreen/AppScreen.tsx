import React from 'react';

import { AppLayout, Button, Input, ResultLabel, UsersList } from '@/components';
import { useGithubRepositoryExplorer } from '@/hooks';

export const AppScreen = () => {
  const {
    usernameValue,
    resultLabel,
    usersByQueryData,
    isLoadingUsersByQueryData,
    isUserListEnabled,
    isResultLabel,
    isValidationError,
    onChangeTextUsernameQuery,
    handleOnSubmitEditingFetchUsersByQuery,
    handleOnPressFetchUsersByQuery,

    repositoriesByUsernameData,
    fetchRepositoriesByUserError,
    isLoadingRepositoriesByUserData,
    userRepositoriesListExpanded,
    onPressUserRepositoriesListExpand,
  } = useGithubRepositoryExplorer();

  return (
    <AppLayout>
      <Input
        value={usernameValue}
        placeholder="Enter username"
        isValidationError={isValidationError}
        isValidationErrorMessage="Username can't be empty"
        onChangeText={onChangeTextUsernameQuery}
        onSubmitEditing={handleOnSubmitEditingFetchUsersByQuery}
      />

      <Button
        label="Search"
        isLoading={isLoadingUsersByQueryData}
        onPress={handleOnPressFetchUsersByQuery}
      />

      {isResultLabel && <ResultLabel resultLabel={resultLabel} />}

      {isUserListEnabled && (
        <UsersList
          users={usersByQueryData?.items}
          userRepositories={repositoriesByUsernameData}
          fetchRepositoriesByUserError={fetchRepositoriesByUserError}
          isLoadingRepositoriesByUserData={isLoadingRepositoriesByUserData}
          userRepositoriesListExpanded={userRepositoriesListExpanded}
          onPressUserRepositoriesListExpand={onPressUserRepositoriesListExpand}
        />
      )}
    </AppLayout>
  );
};
