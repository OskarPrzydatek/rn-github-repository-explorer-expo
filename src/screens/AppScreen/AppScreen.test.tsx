import { render, screen } from '@testing-library/react-native';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

import { AppScreen } from '@/screens/AppScreen/AppScreen';

const queryClient = new QueryClient();

jest.mock('@/hooks', () => ({
  useGithubRepositoryExplorer: () => ({
    isResultLabel: true,
    isUserListEnabled: true,
  }),
}));

describe('AppScreen', () => {
  it('should match snapshot', () => {
    const tree = render(
      <QueryClientProvider client={queryClient}>
        <AppScreen />
      </QueryClientProvider>,
    );

    expect(tree).toMatchSnapshot();
  });

  it('should render result label and users list', () => {
    render(
      <QueryClientProvider client={queryClient}>
        <AppScreen />
      </QueryClientProvider>,
    );

    const resultLabel = screen.getByTestId('app-screen-result-label');
    const usersList = screen.getByTestId('app-screen-users-list');

    expect(resultLabel).not.toBeNull();
    expect(usersList).not.toBeNull();
  });
});
