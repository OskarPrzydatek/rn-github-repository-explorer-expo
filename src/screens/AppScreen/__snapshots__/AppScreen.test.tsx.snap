// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppScreen should match snapshot 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "flex": 1,
      "gap": 16,
      "marginTop": 64,
      "paddingHorizontal": 16,
    }
  }
>
  <View
    style={
      {
        "gap": 8,
      }
    }
  >
    <TextInput
      onBlur={[Function]}
      onFocus={[Function]}
      placeholder="Enter username"
      selectionColor="#1f2329"
      style={
        [
          {
            "borderRadius": 6,
            "height": 40,
            "padding": 12,
          },
          {
            "backgroundColor": "#f6f8fa",
            "borderColor": "#d0d7de",
            "borderWidth": 1,
          },
        ]
      }
    />
  </View>
  <View
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#1e883e",
        "borderRadius": 6,
        "justifyContent": "center",
        "padding": 12,
      }
    }
  >
    <Text
      style={
        {
          "color": "#ffffff",
          "fontWeight": "bold",
        }
      }
    >
      Search
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "baseline",
        "width": "80%",
      }
    }
    testID="app-screen-result-label"
  >
    <View
      style={
        {
          "borderBottomColor": "#fd8c73",
          "borderBottomWidth": 2,
          "paddingBottom": 8,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          {
            "fontWeight": "bold",
          }
        }
      />
    </View>
  </View>
  <RCTScrollView
    contentContainerStyle={
      {
        "gap": 8,
        "paddingBottom": 64,
      }
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={[]}
    testID="app-screen-users-list"
    viewabilityConfigCallbackPairs={[]}
  >
    <View />
  </RCTScrollView>
</RNCSafeAreaView>
`;
